sdk/src/main/java/com/hedera/hashgraph/sdk/Executable.java

```diff
@@ -353,10 +353,13 @@ public O execute(Client client, Duration timeout) throws TimeoutException, Prech
                 throw new MaxAttemptsExceededException(lastException);
             }
 
-            if (Instant.now().isAfter(timeoutTime)) {
+            Duration deadline = Duration.between(Instant.now(), timeoutTime);
+            if (deadline.isNegative() || deadline.isZero()) {
                 throw new TimeoutException();
             }
 
+            this.setGrpcDeadline(deadline);
```

Comment:
There's a `Executable.setGrpcDeadline()` that should still be respected. Here's its value is being overwritten so users can't customize it.